lib/compat/wordpress-5.9/class-wp-rest-edit-site-export-controller.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * REST API: WP_REST_Edit_Site_Export_Controller class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Controller which provides REST endpoint for exporting current templates
+ * and template parts.
+ *
+ * @since 5.9.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Edit_Site_Export_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructs the controller.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp-block-editor/v1';
+		$this->rest_base = 'export';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'export' ),
+					'permission_callback' => array( $this, 'permissions_check' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to export.
+	 *
+	 * @return WP_Error|bool True if the request has access, or WP_Error object.
+	 */
+	public function permissions_check() {
+		if ( current_user_can( 'edit_theme_options' ) ) {
+			return true;
+		}
+
+		return new WP_Error(
+			'rest_cannot_view_url_details',
+			__( 'Sorry, you are not allowed to export templates and template parts.', 'gutenberg' ),
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Output a ZIP file with an export of the current templates
+	 * and template parts from the site editor, and close the connection.
+	 *
+	 * @return WP_Error|void
+	 */
+	public function export() {
+		// Generate the export file.
+		$filename = gutenberg_generate_edit_site_export_file();
```

Comment:
Should `gutenberg_generate_edit_site_export_file` be moved to 5-9 compat and remove the gutenberg prefix? The function is needed to make the endpoint work in core without Gutenberg.

Reply:
Done 👍  But it looks like we still have to use `gutenberg_get_block_templates` inside this function.

Reply:
Looks like `get_block_templates` has was added in 5.8 (https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/block-template-utils.php#L551) and updated in https://core.trac.wordpress.org/changeset/52062

From what I can see `gutenberg_get_block_templates` and `get_block_templates` are identical.

Reply:
I think this is because we have to support at least two versions of WP. The `get_block_templates` will return different results for 5.8 and 5.9. See https://github.com/WordPress/gutenberg/pull/36201#issuecomment-961687771

Reply:
Yes, was just about to add a comment about that 👍  I'll pull down the PR and test it.